input/drivers_hid/iohidmanager_hid.c

```diff
@@ -1105,7 +1105,7 @@ static int32_t iohidmanager_set_report(void *handle, uint8_t report_type, uint8_
       (struct iohidmanager_hid_adapter*)handle;
 
    if (adapter)
-      return IOHIDDeviceSetReport(adapter->handle, translate_hid_report_type(report_type), report_id, data_buf, size);
+      return IOHIDDeviceSetReport(adapter->handle, translate_hid_report_type(report_type), report_type, data_buf, size);
```

Comment:
Do you think it was really designed that way?
The second argument `uint8_t report_type` takes the inner RetroArch's values, that the macOS has no idea about.
https://github.com/libretro/RetroArch/blob/0973e8d4b071d10602b25b5a2271bbec062dcf03/input/include/hid_driver.h#L25-L28

In that case, why is there `report_id` processing inside the `connect_ps3.c` file at all, if it is not used absolutely nowhere?

Reply:
No, I think I'm an idiot. I apologize. I've zapped that commit and done a proper fix for the DS3.